docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/TracerV.rst

```diff
@@ -0,0 +1,280 @@
+
+This is option ``3`` in the ``.ini`` above.
+
+The ``start`` and ``end`` fields are interpreted as hexadecimal values. For
+each, the field is a 64-bit value, with the upper 32-bits representing a
+mask and the lower 32-bits representing a comparison value. That is, the
+start or stop condition will be satisfied when the following evaluates to true:
+
+.. code-block:: C
+
+    ((inst value) & (upper 32 bits)) == (lower 32 bits)
+
+
+That is, setting ``start=ffffffff00008013`` will cause recording to start when
+the instruction value is exactly ``00008013`` (the ``addi x0, x1, 0``
+instruction in RISC-V).
+
+
+This form of triggering is useful when recording traces only when a particular
+application is running within Linux. To simplify the use of this triggering
+mechanism, workloads derived from ``br-base.json`` in FireMarshal automatically
+include the commands ``firesim-start-trigger`` and ``firesim-end-trigger``,
+which issue a ``addi x0, x1, 0`` and ``addi x0, x2, 0`` instruction
+respectively. In your ``config_runtime.ini``, if you set the following
+trigger settings:
+
+.. code-block:: ini
+
+    selector=3
+    start=ffffffff00008013
+    end=ffffffff00010013
+
+
+And then run the following at the bash prompt on the simulated system:
+
+.. code-block:: bash
+
+    $ firesim-start-trigger && ./my-interesting-benchmark && firesim-end-trigger
+
+
+The trace will contain primarily only traces for the duration of
+``my-interesting-benchmark``.  Note that there will be a small amount of extra
+trace information from ``firesim-start-trigger`` and ``firesim-end-trigger``,
+as well as the OS switching between these and ``my-interesting-benchmark``. 
+
+
+.. Attention::  While it is unlikely that a compiler will generate the
+   aforementioned trigger instructions within normal application code, it is also
+   a good idea to confirm that these instructions are not inadvertently present
+   within the section of code you wish to profile. This will result in the trace
+   recording inadvertently turning on and off in the middle of the workload.
+
+   On the flip-side, a developer can deliberately insert the aforementioned ``addi``
+   instructions into the code they wish to profile, to enable more fine-grained
+   control.
+
+
+Interpreting the Trace Result
+------------------------------
+
+Human readable output
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is ``output_format=0``.
+
+The human readable trace output format looks like so:
+
+.. include:: TRACERV-HUMAN-READABLE-EXAMPLE
+   :code: ini
+
+In this output, C0 represents instruction stream (or core) zero. The next field
+represents the address of the instruction committed that cycle and a valid bit,
+in hex, interpreted like so:
+
+.. code-block:: ini
+
+    0000010080000000
+         ^|--------|
+         |       \--- 40 bits of address
+         |
+         \-------- valid bit
+
+
+The final field is the target cycle on which this instruction was committed,
+in hex.
+
+Binary output
```

Comment:
Is there a script somewhere that can post-process the binary format? (@zhemao )

Reply:
discussed in person